### ISSUE TITLE
Running job in Parallel in Windows OS

### ISSUE THREAD

**Issue by PershingWadlow** [firemodels/fds#6756]
I am trying to run a case in FDS by subdividing it into 13 parts. I want to run this job in parallel in Windows OS. I am using this command "mpiexec -n 13 fds jobname.fds"

1. But this is not ruining in parallel in Windows. However running fine in series on the same Windows OS. So I guess my job file is correct. Possible error is in my runscript. 

2. or Does FDS has some maximum no of cells limit in parallel.

Please correct me what I am missing.

Thanks
![error](https://user-images.githubusercontent.com/38319080/43890739-26db0290-9be5-11e8-9355-2ba3c2620867.JPG)

**Comment by rmcdermo** [comment]
Salah can you have a look at this?  Thanks.

**Comment by sbenkorichi** [comment]
@PershingWadlow 
You need to provide further info. It appears that you're having some conflicts. Answer these:
1. Which fds release you're using.
2. on cmd do: `where mpiexec `
3. do : `mpiexec -np 2 -localonly test_mpi `

**Comment by PershingWadlow** [comment]
1. FDS version: FDS6.7.0-SMV6.7.1_win64 
2. where mpiexec :  c:\ Program Files\firemodels\FDS6\bin\mpiexec.exe
3. same error as attached in previous post
Please find info about point 2 and 3 in screenshot also.
![issue](https://user-images.githubusercontent.com/38319080/43896086-3103418c-9bf5-11e8-98c4-66a4e4dd060f.JPG)

**Comment by sbenkorichi** [comment]
Are you installing pyrosim alonside with fds? 

try this `where hydra_service`

**Comment by PershingWadlow** [comment]
Yes I have installed PyroSim in my system before FDS. Please find attached the screenshots for the required info.
Also, when I am running  the same code in series it is working fine but in parallel it is creating the error. Please see the attached error image.

![error](https://user-images.githubusercontent.com/38319080/43910554-c440a37a-9c1a-11e8-8787-3807cb7c55a5.JPG)
![hydra](https://user-images.githubusercontent.com/38319080/43910555-c46d33a4-9c1a-11e8-9b6e-1e834102a2d2.JPG)

**Comment by sbenkorichi** [comment]
My experience is that I have to uninstall pyrosim. Then, install fds, and it should work. 
Try it and report what you get.

**Comment by drjfloyd** [comment]
Ctrl-alt-delete and open the task manager and then click on the services
tab. If both the Pryosim and the Intel mpi Hydra service is running you can
try seeing if stopping the Pyrosim service helps.

**Comment by PershingWadlow** [comment]
After uninstalling pyrosim also same error is repeating.

**Comment by bwklein** [comment]
You could just click to Run Parallel in PyroSim and leave the command
prompt alone.

-Bryan Klein

Best Regards,
-Bryan Klein
klein@thunderheadeng.com
Product Manager, PyroSim/Pathfinder
Thunderhead Engineering Consultants, Inc.
Website - http://www.thunderheadeng.com

**Comment by ankit171032** [comment]
I am also facing the same problem. I tried stopping "Intel mpi Hydra service" and uninstalled pyrosim also. But still getting this hydra error.
Actually I want to run my case in Linux cluster where I cannot install PyroSim so that is why I am trying to run FDS in parallel in Windows. 
Is this not possible to run FDS in parallel in Windows ? and also how to run in parallel in cluster. 
**When I am using the command "mpiexec -n 4 fds jobname.fds" in**

1. Windows : It is creating hydra error and in
2. Linux Cluster : Code is running only in series even with this command. No of MPI processes is always 1. It should be 4 for parallel.

Please suggest some solution.

Thanks

**Comment by sbenkorichi** [comment]
--Bryan, the user is trying to use 6.7. Pyrosim though still with 6.6.

@ankit171032 Can we keep this thread for windows please. Open up new issue explaining the problem you're having with running parallel jobs on Linux and I will take a look.

**Comment by ankit171032** [comment]
@sbenkorichi Ok I will start new thread for linux.
But I am also facing this problem in Windows also. Running FDS in parallel in windows is giving hydra error. 
Please check this problem for current thread.

**Comment by ankit171032** [comment]
Has anyone faced or can suggest how to rectify this hydra error mentioned in previous posts.
Running FDS in parallel in Windows. I tried all the suggestions posted above "uninstalling PyroSim , stopping hydra services" but still the same error.

Please have a look.

Thanks

**Comment by sbenkorichi** [comment]
Open up task manager, go to `Details`, search for `smpd.exe`, select it and then click on End task.
open new `cmd` command, and cd to your fds input file, and rerun the job. 
Let me know how it goes.

**Comment by ankit171032** [comment]
I stopped smpd.exe and now getting the error "credentials rejected". I am the only administrator for my PC.
![capture](https://user-images.githubusercontent.com/31561477/44036274-008d62e8-9f2f-11e8-9fb8-67d3d8f79436.JPG)

**Comment by sbenkorichi** [comment]
add this to your command line: 
`mpiexec -localonly -np 4 fds jobname.fds`

**Comment by ankit171032** [comment]
added 
Now this error
![capture](https://user-images.githubusercontent.com/31561477/44036742-2b6fd0da-9f30-11e8-9a96-67376cc07648.JPG)

**Comment by sbenkorichi** [comment]
Ok, now it's running ok. That's a current issue we're working on. You can follow the update on it here https://github.com/firemodels/fds/issues/6576

Ps: I've submitted a ticket to Intel developers, If they provide a fix, I will comment on that post.

**Comment by ankit171032** [comment]
ok thanks

**Comment by bwklein** [comment]
He can download the windows .exe files and change the FDS binary that
PyroSim uses to the newer version when running the simulation.  This is in
the FDS tab in the PyroSim preferences window.

-Bryan

**Comment by sbenkorichi** [comment]
But that still uses 6.6 mpi libraries. I tried it before and fds6.7 works with it but Kevin said it's quite risky and not recommended as the code wasn't compiled with it, unless you're referring to a different way to get around this, then more of details would be of help.
Thanks !

**Comment by bwklein** [comment]
I'm not sure I understand. Are you using a new version of Intel Hydra process manager (we setup version 5.1.3.0) running as a service?  We use the files in the NIST FDS-SMV bundle, we don't compile them or anything else. If the library files are in the same directory as the fds.exe file, then it should use them when it runs.  Just like running fds from the command line.

**Comment by sbenkorichi** [comment]
I tested your suggestion and I found it still using `mpiexec` that comes with pyrosim (i.e.` 6.6`) instead of the one that comes with the bundle of fds latest (6.7) in the \bin with the `fds.exe`.  
Yes, one can specify the fds executable in the preference tab, but it still by default uses mpi that comes under (`pyrosim 2017\fds\ `) directory for example.  
I believe this issue is due to the update of `Intel Fortran C/C++` and its MPI libraries of Update `18.2`. 
I just compiled a simple test with Update `18.3` and it failed as well. It's like a black magic to get things working under Windows.

**Comment by bwklein** [comment]
It seems we might have a bug there. I'll make an issue for PyroSim to run the fds binary and use mpiexec from the same location when the binary path is changed.

**Comment by sbenkorichi** [comment]
Well, 
I would love to get more feedback on this. This is what I've tried now. 
I copied all `fds6.7\bin` into `pyrosim\fds\` (i.e. I deleted fds6.6. and its mpiexec etc ... and replaced them with latest bundle)
Then run it from pyrosim, it helped 
![image](https://user-images.githubusercontent.com/20113030/44126598-270667c8-a031-11e8-8921-16fa36fe0e64.png)



this is the command that was passed : 

`C:\Program Files\PyroSim 2017\fds\runfds.exe C:\Program Files\PyroSim 2017\fds\mpiexec.exe -n 2 -localonly C:\Program Files\PyroSim 2017\fds\fds.exe C:\Users\SBenkorichi\Desktop\firemodels\fds\Verification\Controls\device_test\device_test.fds`

Using the same command in `cmd `as: 
`"C:\Program Files\PyroSim 2017\fds\runfds.exe" "C:\Program Files\PyroSim 2017\fds\mpiexec.exe" -n 2 -localonly "C:\Program Files\PyroSim 2017\fds\fds.exe" "C:\Users\SBenkorichi\Desktop\firemodels\fds\Verification\Controls\device_test\device_test.fds"`

It fails to work, do you have any idea why it's failing though it's same command. What am I missing here?

**Comment by sbenkorichi** [comment]
In the latest release (6.7.1), this issue has disappeared. 
Report back if it's not working for you.
 ```
mpiexec -localonly -np 2 fds device_test.fds
 Reading input file ...

 Fire Dynamics Simulator

 Current Date     : February  6, 2019  15:17:45
 Revision         : FDS6.7.1-0-g14cc738-master
 Revision Date    : Mon Feb 4 12:26:25 2019 -0500
 Compiler         : Intel ifort 19.0.1.144
 Compilation Date : Mon 02/04/2019  01:41 PM

 MPI Enabled;    Number of MPI Processes:       2
 OpenMP Enabled; Number of OpenMP Threads:      4

 MPI version: 3.0
 MPI library version: Intel(R) MPI Library 5.1.3 for Windows* OS


 Job TITLE        : Test sprinklers, heat detectors and smoke detectors
 Job ID string    : device_test

 Time Step:      1, Simulation Time:      0.16 s
 Time Step:      2, Simulation Time:      0.32 s
 Time Step:      3, Simulation Time:      0.47 s
 Time Step:      4, Simulation Time:      0.63 s
 Time Step:      5, Simulation Time:      0.72 s
 Time Step:      6, Simulation Time:      0.78 s
 Time Step:      7, Simulation Time:      0.84 s
 Time Step:      8, Simulation Time:      0.90 s
 Time Step:      9, Simulation Time:      0.95 s
 Time Step:     10, Simulation Time:      0.98 s
 Time Step:     20, Simulation Time:      1.21 s
 Time Step:     30, Simulation Time:      1.39 s
 Time Step:     40, Simulation Time:      1.58 s
 Time Step:     50, Simulation Time:      1.79 s
 Time Step:     60, Simulation Time:      2.00 s
 Time Step:     70, Simulation Time:      2.18 s
```
